File: model/Pocket.java
package model;

import java.util.ArrayList;
import java.util.List;
import model.common.PocketObject;

public class Pocket {
  private final List<PocketObject> objects;
  private final int number;

  public Pocket(int number) {
    this.objects = new ArrayList<>();
    this.number = number;
  }

  public List<PocketObject> getObjects() {
    return objects;
  }

  public int getNumber() {
    return number;
  }

  public void empty() {
    objects.clear();
  }

  public void putObject(PocketObject object){
    objects.add(object);
  }

  public void retrieveObject(PocketObject object){
    objects.remove(object);
  }


  @Override
  public String toString() {
    return "Pocket{" +
        "objects=" + objects +
        ", number=" + number +
        '}';
  }
}

File: model/Pocket.py
from typing import List

from model.common import PocketObject


class Pocket:
    def __init__(self, number):
        self._objects: List[PocketObject] = []
        self._number = number

    def get_objects(self):
        return self._objects

    def get_number(self):
        return self._number

    def empty(self):
        self._objects.clear()

    def put_object(self, obj):
        self._objects.append(obj)

    def retrieve_object(self, obj):
        self._objects.remove(obj)

    def __str__(self):
        return f"Pocket{{objects={self._objects}, number={self._number}}}"
